lib/electric_slide/call_queue.rb

```diff
@@ -164,7 +164,7 @@ def add_agent(agent)
       abort ArgumentError.new("#add_agent called with nil object") if agent.nil?
       abort DuplicateAgentError.new("Agent is already in the queue") if get_agent(agent.id)
 
-      agent.queue = self
```

Comment:
For my education, can you explain the difference between `self` here (which should be the current instance?) and `current_actor`?